client/blocks/google-my-business-stats-nudge/index.js

```diff
@@ -0,0 +1,116 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+import { flow } from 'lodash';
+
+/**
+ * Internal Dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import { recordTracksEvent } from 'state/analytics/actions';
+import SectionHeader from 'components/section-header';
+import QueryPreferences from 'components/data/query-preferences';
+import { isGoogleMyBusinessStatsNudgeDismissed } from 'state/selectors';
+import { dismissNudge } from './actions';
+
+class GoogleMyBusinessStatsNudge extends Component {
+	static propTypes = {
+		siteSlug: PropTypes.string.isRequired,
+		trackNudgeAlreadyListedClick: PropTypes.func.isRequired,
+		trackNudgeDismissClick: PropTypes.func.isRequired,
+		trackNudgeStartNowClick: PropTypes.func.isRequired,
+		trackNudgeView: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	componentWillMount() {
+		if ( ! this.props.isDismissed ) {
+			this.props.trackNudgeView();
+		}
+	}
+
+	render() {
+		if ( this.props.isDismissed ) {
```

Comment:
I'm a little bothered with the logic of deciding to show or not show the nudge to be split between:
`this.props.isDismissed` => `isGoogleMyBusinessStatsNudgeDismissed( state )`
which checks dismissal times and how long ago was that dismiss and [isGoogleMyBusinessStatsNudgeVisible](https://github.com/Automattic/wp-calypso/blob/master/client/state/selectors/is-google-my-business-stats-nudge-visible.js#L50) which is called from the [upper level](https://github.com/Automattic/wp-calypso/blob/c02130167fd3a9aa4e025159bc75f94b224dd789/client/my-sites/stats/site.jsx#L146-L149) and checks how long ago a site was created along with what goals it has.

IMO those 3 conditions belong together:
1. `isGoogleMyBusinessStatsNudgeVisibleSelector(
				state,
				siteId
			)`
2. `config.isEnabled( 'google-my-business' )`
3. `isGoogleMyBusinessStatsNudgeDismissed( state )`


Also, another related question: Why do we ask if display the nudge according to a specific site ( `isGoogleMyBusinessStatsNudgeVisibleSelector` ), but we count dismisses globally ( `isGoogleMyBusinessStatsNudgeDismissed` ) ?